lib/src/components/NxSkeletonLoader/NxSkeletonLoader.tsx

```diff
@@ -0,0 +1,110 @@
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import { IconProp } from '@fortawesome/fontawesome-svg-core';
+import { find } from 'ramda';
+import React, { createContext, HTMLAttributes, ReactNode, useEffect } from 'react';
+import withClass from '../../util/withClass';
+import NxFontAwesomeIcon from '../NxFontAwesomeIcon/NxFontAwesomeIcon';
+
+import './NxSkeletonLoader.scss';
+
+export const SkeletonContext = createContext(false);
+
+const ANIMATION_NAME = 'skeleton-animation';
+
+// These two variables help globally synchronize skeleton loader animation state. Whenever a skeleton loader
+// is present on the page, any additional skeleton loaders should synchronize with it. Once all skeleton loaders
+// are removed, that synchronization state should be cleared so that a future skeleton loader added later will
+// start its animation cleanly at the beginning and not based on the original skeleton loader's time
+const activeSkeletonLoaders = new Set();
+let animationStartTime: number | null = null;
+
+/*
+ * According to the spec, animations have an id property (that is presumably supposed
+ * to be the CSS keyframe name) and not an animationName property. However, at the time of writing,
+ * in at least Chrome and firefox the id for our animation is the empty string and an `animationName` property
+ * contains the name of the CSS keyframes
+ */
+function isSkeletonAnimation(animation: Animation) {
+  return animation.id === ANIMATION_NAME || (animation as { animationName?: string }).animationName === ANIMATION_NAME;
+}
+
+function _NxSkeletonLoader({ children }: { children: ReactNode }) {
+  /*
+   * Synchronize all skeleton animations
+   */
+  useEffect(() => {
+    // a unique object to identify this NxSkeletonLoader instance in the activeSkeletonLoaders set
+    const instance = {};
+
+    function handleAnimationStart({ animationName, target }: AnimationEvent) {
+      if (animationName === ANIMATION_NAME) {
+        const animation = target instanceof Element && find(isSkeletonAnimation, target.getAnimations());
+
+        if (animation) {
+          if (animationStartTime) {
+            animation.startTime = animationStartTime;
+          }
+          else {
+            animationStartTime = animation.startTime;
+          }
+        }
+      }
+    }
+
+    if (!activeSkeletonLoaders.size) {
+      window.addEventListener('animationstart', handleAnimationStart);
+    }
+
+    activeSkeletonLoaders.add(instance);
+
+    return () => {
+      activeSkeletonLoaders.delete(instance);
+      if (!activeSkeletonLoaders.size) {
+        animationStartTime = null;
+        window.removeEventListener('animationstart', handleAnimationStart);
+      }
+    };
+  }, []);
+
+  return (
+    <div className="nx-skeleton-loader" aria-busy="true">
```

Comment:
In a real impl, `NxSkeletonLoader` would function more like `NxLoadWrapper`.  It would receive a `loading` boolean prop and would only enable its loading state if that prop is true.  i.e., `aria-busy` here would be set conditionally, as would `aria-hidden` below an the context provider value.  The `.nx-skeleton-loader__status` would also be conditionally hidden from the screenreader.

Reply:
> In a real impl, NxSkeletonLoader would function more like NxLoadWrapper. It would receive a loading boolean prop and would only enable its loading state if that prop is true. i.e., aria-busy here would be set conditionally, as would aria-hidden below an the context provider value. The .nx-skeleton-loader__status would also be conditionally hidden from the screenreader. For the spike I left that out and just have it always active

Also like `NxLoadWrapper`, this component will need to handle load errors, which it should do by rendering an error alert instead of content.